### ISSUE TITLE
Updating list of open-access DOI prefixes?

### ISSUE THREAD

**Issue by a3nm** [dissemin/dissemin#728]
Our list of open-access DOI prefixes in `dissemin/backend/doiprefixes.py` comes from https://github.com/Impactstory/oadoi/blob/ce1e51109151674e9a6c69b448cebacc9c19eb1e/oa_local.py. Of course, this source can evolve independently: I have just went over what they did up to now and reflected it (deleting one prefix).

The source for this file is http://stats.datacite.org/ which itself might evolve. Guessing from the network requests done on this page, I think I can retrieve programmatically the list of prefixes with `curl 'https://search.datacite.org/list/generic?&facet.field=prefix' | cut -d';' -f1`. There are 1640 options, vs 661 that we currently know (and 25 that we blacklisted).

**Do you think it would be a good idea to update `doiprefixes.py` with these prefixes (except the blacklisted ones)?.** The thing is, I don't know what's the quality of this data or what it means for a prefix to be listed there.

A related question is whether it would make sense to learn our own classifier based on DOI prefixes? (i.e., learn which DOI prefixes are OA and which are not by crawling a few in each and seeing if we can access the PDF). Would you expect this to work better than URL-based patterns, or worst?

(Also, I see that apparently `free_doi_prefixes` is only checked on paper creation, so changes to it, either when adding or deleting, would need to be ported to existing publications...)

**Comment by wetneb** [comment]
Yes, updating this data is a good idea. By the way, it might be worth checking if any of the false positives/negatives from https://github.com/dissemin/dissemin/issues/602 are due to these DOI prefixes.

If many publications are affected by this, it is possible to write a migration that will mass-update their status once, for specified DOI prefixes.

**Comment by a3nm** [comment]
(Just checked and none of the false positives in #602 is caused by `free_doi_prefixes`.)
